kafka-connect-s3/src/main/java/io/confluent/connect/s3/storage/S3OutputStream.java

```diff
@@ -175,6 +182,64 @@ private MultipartUpload newMultipartUpload() throws IOException {
     }
   }
 
+  protected static class RetryWithExponentialBackoff {
+    private int failCount;
+    private int maxRetryAttempts;
```

Comment:
I believe we need one to keep track of current position and another as a boundary. Having only one counter and decrementing to zero would make it harder(or impossible?) to implement exponential back off. However as I removed the class so these are not class fields any longer.